src/Entity/User.php

```diff
@@ -189,6 +189,14 @@ public function getRoles(): array
         return array_unique($roles);
     }
 
+    /**
+     * @param string[] $roles
+     */
+    public function changeRoles(array $roles): void
+    {
+        $this->roles = array_unique($roles);
```

Comment:
Should add array_values here to prevent the serialized value to transform into json object 🤷